geopy/geocoders/banfrance.py

```diff
@@ -0,0 +1,194 @@
+import warnings
+
+from geopy.compat import urlencode
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import logger
+
+__all__ = ("BANFrance", )
+
+
+class BANFrance(Geocoder):
+    """Geocoder using the Base Adresse Nationale France API.
+
+    Documentation at:
+        https://adresse.data.gouv.fr/api
+
+    """
+
+    geocode_path = '/search'
+    reverse_path = '/reverse'
+
+    def __init__(
+            self,
+            domain='api-adresse.data.gouv.fr',
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            format_string=None,
+            ssl_context=DEFAULT_SENTINEL,
+    ):
+        """
+
+        :param str domain: Currently it is ``'api-adresse.data.gouv.fr'``, can
+            be changed for testing purposes.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+            .. versionadded:: 1.12.0
+
+        :param str format_string:
+            See :attr:`geopy.geocoders.options.default_format_string`.
+
+            .. versionadded:: 1.14.0
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+            .. versionadded:: 1.14.0
```

Comment:
Could you please remove these version-related directives? This class didn't even exist in these versions, so these arguments couldn't have been technically added to this class in these versions.